ui: Tooltips for statements list page

vladlos opened the pull request:
Added column header tooltips statements table and for diagnostic statuses

![TXN TYPE](https://user-images.githubusercontent.com/12850886/77671645-97254f80-6f90-11ea-99c0-ea081ab76243.png)

![DIAGNOSTICS](https://user-images.githubusercontent.com/12850886/77671703-a73d2f00-6f90-11ea-9737-3dc4d579a5cd.png)

Resolves: #46211, #48075

Release justification: low risk, high benefit changes to existing functionality

Release note (ui): statements tooltips design updates

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46626)
<!-- Reviewable:end -->

vladlos committed:
ui: Tooltips for statements list page

Added column header tooltips statements table and for diagnostic statuses

Resolves: #46211

Release justification: low risk, high benefit changes to existing functionality

Release note (ui): statements tooltips design updates

dhartunian reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46626)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @elkmaster)

___
*[pkg/ui/src/views/statements/statementsTable.tsx, line 26 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-M4R03vt8f7pKpAYof-N:-M4R03vu7rwAZCBDxE5z:b-z01497) ([raw file](https://github.com/cockroachdb/cockroach/blob/ee0eac7c5cc2b9228b12fb2babe0932d053f10fe/pkg/ui/src/views/statements/statementsTable.tsx#L26)):*
> ```XML
> import IStatementDiagnosticsReport = cockroach.server.serverpb.IStatementDiagnosticsReport;
> import { ActivateDiagnosticsModalRef } from "./diagnostics/activateDiagnosticsModal";
> import { statementsSql, transactionalPipelining, statementDiagnostics, statementsRetries, statementsTimeInterval } from "oss/src/util/docs";
> ```

remove `oss` from import
___
*[pkg/ui/src/views/statements/statementsTable.tsx, line 85 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-M4R05ng-tfSItE7W-WQ:-M4R05nh9UVptNgM_2Oi:b87tjev) ([raw file](https://github.com/cockroachdb/cockroach/blob/ee0eac7c5cc2b9228b12fb2babe0932d053f10fe/pkg/ui/src/views/statements/statementsTable.tsx#L85)):*
> ```XML
>     {
>       title: (
>         <Tooltip
> ```

Extract these tooltip `title` props and give them names in a separate file. They pollute these already large files and make them difficult to read.


<!-- Sent from Reviewable.io -->

dhartunian commented:
@elkmaster and I discussed in-person that this PR is a good candidate to try an internationalization (i18n) library to keep track of all the text snippets.

We'll use this PR to prototype this approach. Looking at possibly using https://react.i18next.com/.

dhartunian reviewed (changes requested):
Thanks @elkmaster for installing the i18n library and integrating it with the app!

- since you're adding a new dependency, can you list the PR on `yarn-vendored` that we need to merge with this?
- can you provide documentation in the commit message on usage and meaning of `<1>, <2>, etc.` tags?

Do we have any alternatives to doing the numbered tags if we want to use HTML inside the templates? Seems like all the tags you're using are: `code`, `br`, and `Anchor`. Could we just allow unescaped HTML to make things simpler for now since the inputs is fully under our control. Instead of `Anchor` we can just use `<a>` for now.
What do you think?

> Reviewed 5 of 5 files at r1, 1 of 3 files at r3, 3 of 7 files at r5, 5 of 7 files at r6.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-:-M5hwcgUBEhziu1SvXzf:b-o8ag1d)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian and @elkmaster)

___
*[pkg/ui/src/views/statements/statementsTable.tsx, line 75 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-M5i0bfE5Gy_ZkOkIo2w:-M5i0bfE5Gy_ZkOkIo2x:b-nniizh) ([raw file](https://github.com/cockroachdb/cockroach/blob/53f2013c6be4350bccd60b27a9fcb739821779fe/pkg/ui/src/views/statements/statementsTable.tsx#L75)):*
> ```XML
>     const diagnosticsColumn: ColumnDescriptor<AggregateStatistics> = {
>       title: StatementTableTitle().diagnostics,
>       cell: StatementTableCell.diagnostics(activateDiagnosticsRef),
> ```

Why not just makes this a functional component?
___
*[pkg/ui/src/views/statements/statementsTable.tsx, line 128 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-M5hxqyp-pMbvaTWwWNq:-M5hxqyq5ybROkFeIMDD:bmohfrg) ([raw file](https://github.com/cockroachdb/cockroach/blob/53f2013c6be4350bccd60b27a9fcb739821779fe/pkg/ui/src/views/statements/statementsTable.tsx#L128)):*
> ```XML
>     },
>     {
>       title: StatementTableTitle().latency,
> ```

you're re-evaluating the `StatementTableTitle` function many times in this file. why is this necessary? Why not make it a functional component?


<!-- Sent from Reviewable.io -->

vladlos reviewed (commented):
<1>, <2>, etc. is a way how this library interpolate layout in translation string. basically you pass array of tags, that will be used in <%> by index. this is documented here https://react.i18next.com/latest/trans-component. there is security concern in term of putting tags in translate string directly and using `dangerouslySetInnerHTML`. 

I assume if there is not a lot of tooltips will have markdown, we can live with that <123>, or just split string. So translation file will look something like this
`{t1: "SQL statement"
t2: "fingerprint"
t3: "To view additional details of a SQL statement fingerprint, click this to open the Statement Details page."}`

instead of 

`"Cumulative number of <1>retries</1> of statements with this fingerprint within the last hour or specified time interval."`

What do you think?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-:-M69l9JG6g21bQTJIf76:b-5c487i)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian)

___
*[pkg/ui/src/views/statements/statementsTable.tsx, line 26 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-M4R03vt8f7pKpAYof-N:-M69l0KWCLouxcZIZmlB:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/ee0eac7c5cc2b9228b12fb2babe0932d053f10fe/pkg/ui/src/views/statements/statementsTable.tsx#L26)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

remove `oss` from import
</blockquote></details>

Done.
___
*[pkg/ui/src/views/statements/statementsTable.tsx, line 75 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-M5i0bfE5Gy_ZkOkIo2w:-M69kT-f5tPMH1huCCOg:be478ya) ([raw file](https://github.com/cockroachdb/cockroach/blob/53f2013c6be4350bccd60b27a9fcb739821779fe/pkg/ui/src/views/statements/statementsTable.tsx#L75)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Why not just makes this a functional component?
</blockquote></details>

there is 3 functional components underneath, grouped in 1 json for smaller imports
___
*[pkg/ui/src/views/statements/statementsTable.tsx, line 128 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-M5hxqyp-pMbvaTWwWNq:-M69kKAWB1uiZ3BQ8Mtd:b-kwdxp) ([raw file](https://github.com/cockroachdb/cockroach/blob/53f2013c6be4350bccd60b27a9fcb739821779fe/pkg/ui/src/views/statements/statementsTable.tsx#L128)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

you're re-evaluating the `StatementTableTitle` function many times in this file. why is this necessary? Why not make it a functional component?
</blockquote></details>

converted to simple json, no need to execute


<!-- Sent from Reviewable.io -->

dhartunian reviewed (changes requested):
A bunch of files need copyright header.

> Reviewed 2 of 12 files at r7, 3 of 7 files at r9, 5 of 8 files at r10.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-:-M8GCjd12swA5PnBa7-o:b-2kck1g)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian and @elkmaster)

___
*[pkg/ui/src/views/statements/statemetsTableContent.tsx, line 0 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-M8Gd2i2EFKjZyfsgYaQ:-M8Gd2i2EFKjZyfsgYaR:b-sf644q) ([raw file](https://github.com/cockroachdb/cockroach/blob/51f2abcec7d0d351d475c37034d98407a2c4deec/pkg/ui/src/views/statements/statemetsTableContent.tsx#L0)):*
After thinking about this more I've come to the conclusion that the i18n experiment is not worth keeping. Happy to hear your thoughts but I think the cryptic formatting combined with the tags in the `json` will make it *more* difficult to edit the text, not less.

Can you inline the text from the `json` into this file and remove the i18n integration? I think that will make the tooltips easier to manage without introducing too much complication. The amount of extra formatting and linking we need to do in the tooltips makes me think we should just have them in this file.

Also: this filename has a spelling error, it should be: `statementsTableContent.tsx`.


<!-- Sent from Reviewable.io -->

vladlos committed:
ui: Statements table tooltips

Move tooltips contents in separate file

Release note (ui): none

dhartunian reviewed (approved):
Thanks for trying out the i18n option, much appreciated @elkmaster.

This looks good!

> Reviewed 7 of 11 files at r11, 2 of 3 files at r12.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46626#-:-M8QO7E18yMLjJV8sKxJ:b-bhzhcf)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian and @elkmaster)



<!-- Sent from Reviewable.io -->

dhartunian commented:
bors r+
